pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -13,50 +13,50 @@ import (
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/client-go/kubernetes"
 	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
 	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog"
 
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
 	"github.com/openshift/cluster-etcd-operator/pkg/operator/operatorclient"
 )
 
 // EtcdEndpointsController maintains a configmap resource with
 // IP addresses for etcd. It should never depend on DNS directly or transitively.
 type EtcdEndpointsController struct {
-	operatorClient  v1helpers.OperatorClient
+	operatorClient  v1helpers.StaticPodOperatorClient
 	nodeLister      corev1listers.NodeLister
 	configmapLister corev1listers.ConfigMapLister
 	configmapClient corev1client.ConfigMapsGetter
 }
 
 func NewEtcdEndpointsController(
-	operatorClient v1helpers.OperatorClient,
+	operatorClient v1helpers.StaticPodOperatorClient,
 	eventRecorder events.Recorder,
 	kubeClient kubernetes.Interface,
 	kubeInformers operatorv1helpers.KubeInformersForNamespaces,
 ) factory.Controller {
-	kubeInformersForTargetNamespace := kubeInformers.InformersFor(operatorclient.TargetNamespace)
-	configmapsInformer := kubeInformersForTargetNamespace.Core().V1().ConfigMaps()
-	kubeInformersForCluster := kubeInformers.InformersFor("")
-	nodeInformer := kubeInformersForCluster.Core().V1().Nodes()
+	nodeInformer := kubeInformers.InformersFor("").Core().V1().Nodes()
 
 	c := &EtcdEndpointsController{
 		operatorClient:  operatorClient,
 		nodeLister:      nodeInformer.Lister(),
-		configmapLister: configmapsInformer.Lister(),
+		configmapLister: kubeInformers.ConfigMapLister(),
 		configmapClient: kubeClient.CoreV1(),
 	}
 	return factory.New().ResyncEvery(time.Minute).WithInformers(
 		operatorClient.Informer(),
-		configmapsInformer.Informer(),
+		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().ConfigMaps().Informer(),
```

Comment:
Curious what is the difference between the former and the latter?

Reply:
functionally equivalent, both cause sync to trigger on configmap change in the openshift-etcd namespace

Reply:
OK so why change :)

Reply:
Because configmapsInformer was only used in one place (so what's the point of a local to begin with), and also had a misleading name (it isn't an informer for all configmaps, it's only for the openshift-etcd namespace).